### PR TITLE
[NPUW] Better utilize threads on compilation

### PR DESCRIPTION
**Pull request by smirnov-alexey**

### CODE REVIEW

File: src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp [hunk 1]
```diff
@@ -322,13 +322,26 @@ ov::npuw::CompiledModel::CompiledModel(const std::shared_ptr<ov::Model>& model,
     std::map<std::size_t, std::string> forced_sub_devices{};
     const std::string fsd_opt = m_cfg.get<::intel_npu::NPUW_SUBMODEL_DEVICE>();
     forced_sub_devices = ::intel_npu ::OptionParser<std::map<std::size_t, std::string>>::parse(fsd_opt);
+
+    // Exclude optimized out subgraphs from compilation target beforehand - otherwise we might get head and repeated
+    // block in the same chunk
+    std::vector<std::size_t> idx_subgraph_to_compile;
+    for (std::size_t i = 0u; i < orderedSubgraphs.size(); i++) {
+        if (orderedSubgraphs[i]._optimized_out ||
+            (m_compiled_submodels[i].replaced_by && m_compiled_submodels[i].replaced_by != i)) {
```
Comment (dmatveev) [inline src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp:331]:
```suggestion
            m_compiled_submodels[i].replaced_by.value_or(i) != i) {
```

File: src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp [hunk 2]
```diff
@@ -384,14 +397,27 @@ ov::npuw::CompiledModel::CompiledModel(const std::shared_ptr<ov::Model>& model,
...
     }
 
+    // Process skipped submodels
+    for (std::size_t i = 0u; i < orderedSubgraphs.size(); i++) {
+        if (orderedSubgraphs[i]._optimized_out) {
+            continue;  // do nothing here
+        }
+
+        const std::size_t real_id = m_compiled_submodels[i].replaced_by.value_or(i);
+        if (m_compiled_submodels[i].replaced_by && real_id != i) {
```
Comment (dmatveev) [inline src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp:415]:
```suggestion
        const std::size_t real_id = m_compiled_submodels[i].replaced_by.value_or(i);
        if (real_id != i) {
```

File: src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp [hunk 3]
```diff
@@ -384,14 +397,27 @@ ov::npuw::CompiledModel::CompiledModel(const std::shared_ptr<ov::Model>& model,
...
+            m_compiled_submodels[i].compiled_model = m_compiled_submodels[real_id].compiled_model;
+            m_compiled_submodels[i].device_it = m_compiled_submodels[real_id].device_it;
```
Comment (dmatveev) [inline src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp:417]: we didn't do this before. Why do we do this now?

**Review by dmatveev — CHANGES_REQUESTED** [review]

File: src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp (hunk 1 above)
Comment (smirnov-alexey) [reply src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp]: Done

File: src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp (hunk 2 above)
Comment (smirnov-alexey) [reply src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp]: Done

File: src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp (hunk 3 above)
Comment (smirnov-alexey) [reply src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp]: Removed, thanks

Comment (dmatveev) [reply src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp]: Does it work now? :)

**Review by dmatveev — APPROVED** [review]

File: src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp (hunk 3 above)
Comment (smirnov-alexey) [reply src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp]: It does!

**Comment by dmatveev** [comment]
PyTorch tests keep failing. I gave up restarting these builds.

**Pull request merged**